run_api.py

```diff
@@ -3,11 +3,23 @@
 
 import uvicorn
 from fastapi import FastAPI
```

Comment:
let's refactor the API a but for a "separation of concerns"

You can ask cursor to give you a better file structure.